test/unit/scan/test_folder_traversal.py

```diff
@@ -654,3 +654,56 @@ def test_folder_all_files__dir_excluded_by_regex(self, tmp_path):
         assert absolute_paths == [
             fix_windows_path_limit(str(d1_dir / "file1.txt")),
         ]
+
+    def test_excluded_folder_no_access_check(self, tmp_path):
+        """Test that a directory is not checked for access if it is excluded."""
+        # Create directories and files
+        excluded_dir = tmp_path / "excluded_no_access"
+        excluded_dir.mkdir()
+        excluded_file = excluded_dir / "should_not_access.txt"
+        excluded_file.touch()
+
+        # Setup exclusion regex that matches the directory name
+        scan_policy = ScanPoliciesManager(exclude_dir_regexes=[r"excluded_no_access$"])
+        reporter = ProgressReport(sys.stdout, False)
+
+        # Patch os.access to monitor if it is called on the excluded file
+        with patch('os.access', MagicMock(return_value=True)) as mocked_access:
+            folder = LocalFolder(str(tmp_path))
+            list(folder.all_files(reporter=reporter, policies_manager=scan_policy))
+
+            # Verify os.access was not called for the excluded file
+            mocked_access.assert_not_called()
+
+        reporter.close()
+
+    def test_excluded_folder_without_permissions(self, tmp_path):
```

Comment:
what about filtering out the files? since the logic executed is slightly different they deserve they own  test case
or - and I'm totally on board with that - have these test cases tests dir and file cases in single tests

whatever you think is right, but files should be tested as well